src/libs/Navigation/AppNavigator/MainDrawerNavigator.js

```diff
@@ -100,7 +104,15 @@ class MainDrawerNavigator extends Component {
         if (!this.initialParams.reportID) {
             const state = this.props.navigation.getState();
             const reportScreenKey = lodashGet(state, 'routes[0].state.routes[0].key', '');
-            Navigation.setParams(initialNextParams, reportScreenKey);
+
+            // If this onyx key exists then it means we stashed the url from a deep link
+            // so we want to override the reportID that is being loaded by default to the one the user was attempting to load initially.
+            const reportScreenReportID = this.props.reportScreenReportID.split('/r/')[1];
```

Comment:
I don't think it will work in following cases:

- Go to a room > Details > Use the URL to sign in again
- Go to a group > Participants> Use the url to sign in again

The format of url in these cases: /r/reportID/details and /r/reportID/participants

Reply:
Hmm yeah good point. I just tested on staging to see what would happen and this is how it appears right now 

<img width="1512" alt="image" src="https://user-images.githubusercontent.com/4073354/233746254-7b1241e5-9baa-4b19-b65d-82081b3d8841.png">

I would test it but I imagine as Sobit said that this doesn't handle that well. I wonder if we can add the `details` part into the `initialNextParams` object as well to get it to load the details or participants drawers?

Reply:
@bondydaa @sobitneupane I think we could fix to check if `reportScreenReportID` only have number update `initialNextParams.reportID` to `reportScreenReportID`, else update it to `reportScreenReportID.split('/')[0]`. Howerver, with url has format : /r/reportID/details or /r/reportID/participants before load exact it will display "That page is nowhere to be found" same as below because It need to load report before load data in this.


https://user-images.githubusercontent.com/129500732/233948002-d1077906-521a-4d94-8b77-d6092682f4fb.mp4

Reply:
As url already has reportID, instead of recent chat, user should be redirected to report with the reportID in url.

Reply:
@sobitneupane So summary, with url has reportID, user should be redirected to current report with reportID in url and don't show modal detail or participants or some else

Reply:
Nope. I think we should show the detail or participants as well, if the url contains '/details' or '/participants'.

Reply:
@sobitneupane Thanks, I got it. And Is screenshot below the result that you expected?


https://user-images.githubusercontent.com/129500732/234177327-306bc5dc-e623-4eb2-b965-a881a2d46934.mp4

Reply:
Yeah. It is the expected output.

Reply:
@sobitneupane I updated the PR same as your review, please help to check again